File: apps/dotnet/Functions/SlowResponseFunction.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.Functions.Worker;
using Microsoft.Extensions.Logging;

namespace AzureFunctionsGuide.Functions;

/// <summary>
/// Slow response endpoint — delays response by configurable duration for latency testing.
/// Route: GET /api/slow?delay=N
/// </summary>
public class SlowResponseFunction
{
    private readonly ILogger<SlowResponseFunction> _logger;

    public SlowResponseFunction(ILogger<SlowResponseFunction> logger)
    {
        _logger = logger;
    }

    [Function("slowResponse")]
    public async Task<IActionResult> Run(
        [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "slow")] HttpRequest req)
    {
        var delayStr = req.Query["delay"].FirstOrDefault() ?? "2";
        if (!int.TryParse(delayStr, out var delaySec))
            delaySec = 2;

        _logger.LogInformation("Slow response: delaying {Seconds}s", delaySec);

        await Task.Delay(delaySec * 1000);

        var result = new
        {
            delayed = delaySec,
            message = $"Response after {delaySec} second(s)"
        };

        return new OkObjectResult(result);
    }
}

File: apps/java/src/main/java/com/functions/SlowResponseFunction.java
package com.functions;

import com.microsoft.azure.functions.*;
import com.microsoft.azure.functions.annotation.*;
import java.util.Optional;

/**
 * Slow response endpoint — delays response by configurable duration for latency testing.
 */
public class SlowResponseFunction {

    @FunctionName("slowResponse")
    public HttpResponseMessage run(
            @HttpTrigger(
                name = "req",
                methods = {HttpMethod.GET},
                authLevel = AuthorizationLevel.ANONYMOUS,
                route = "slow")
            HttpRequestMessage<Optional<String>> request,
            final ExecutionContext context) {

        String delayParam = request.getQueryParameters().getOrDefault("delay", "2");
        int delaySec;
        try {
            delaySec = Integer.parseInt(delayParam);
        } catch (NumberFormatException e) {
            delaySec = 2;
        }

        context.getLogger().info("Slow response: delaying " + delaySec + "s");

        try {
            Thread.sleep(delaySec * 1000L);
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }

        String body = String.format(
            "{\"delayed\":%d,\"message\":\"Response after %d second(s)\"}",
            delaySec, delaySec);

        return request.createResponseBuilder(HttpStatus.OK)
                .header("Content-Type", "application/json")
                .body(body)
                .build();
    }
}
